ckcp/gitops.sh

```diff
@@ -153,97 +159,155 @@ install_openshift_gitops() {
-    echo -n "."
-    sleep 2
-  done
-  echo "OK"
-
   # Register the host cluster to KCP
-  echo -n "  - Workloadcluster pipeline-cluster registration: "
-  if ! KUBECONFIG="$KUBECONFIG_KCP" oc get cluster local >/dev/null 2>&1; then
-    (
-      curl --fail --silent https://raw.githubusercontent.com/kcp-dev/kcp/948dbe9565cc7da439c698875ca1fa78350c4530/contrib/examples/cluster.yaml
-      sed -e 's/^/    /' "$KUBECONFIG"
-    ) >"$kube_dir/cluster.yaml"
-    KUBECONFIG="$KUBECONFIG_KCP" oc apply -f "$kube_dir/cluster.yaml" >/dev/null
-    rm "$kube_dir/cluster.yaml"
+  echo -n "  - Workspace: "
+  # It's not allowed to create WorkloadCluster resource in a non-private workspace
+  if ! KUBECONFIG="$KUBECONFIG_KCP" kubectl get workspaces demo >/dev/null 2>&1; then
+    KUBECONFIG="$KUBECONFIG_KCP" kubectl kcp workspaces create demo --enter &>/dev/null
   fi
   echo "OK"
 
-  echo -n "  - Workspace: "
-  if ! KUBECONFIG="$KUBECONFIG_KCP" oc get workspaces demo >/dev/null 2>&1; then
-    KUBECONFIG="$KUBECONFIG_KCP" oc create -f "$SCRIPT_DIR/../workspace.yaml" &>/dev/null
+  echo -n "  - Workloadcluster pipeline-cluster registration: "
+  if ! KUBECONFIG="$KUBECONFIG_KCP" kubectl get WorkloadCluster local >/dev/null 2>&1; then
+    (
+      KUBECONFIG="$KUBECONFIG_KCP" kubectl kcp workload sync local  --syncer-image ghcr.io/kcp-dev/kcp/syncer:release-0.4  > "$kube_dir/syncer.yaml"
+      kubectl apply -f "$kube_dir/syncer.yaml" --wait >/dev/null
+      rm -rf "$kube_dir/syncer.yaml"
+    )
   fi
   echo "OK"
 
   # Register the KCP cluster into ArgoCD
   echo -n "  - KCP cluster registration to ArgoCD: "
   if ! KUBECONFIG="$KUBECONFIG_MERGED" argocd cluster get kcp >/dev/null 2>&1; then
-
-    # <workaround> See https://github.com/kcp-dev/kcp/issues/535
-    # Manually create the serviceaccount and secret on the KCP cluster for ArgoCD to use
-    local argocd_yaml="$kube_dir/argocd_kcp.yaml"
-    local argocd_sa="argocd-manager"
-    local argocd_ns="kube-system"
-    echo "---
-apiVersion: v1
-kind: Secret
-metadata:
-  name: $argocd_sa
-  namespace: $argocd_ns
-  annotations:
-    kubernetes.io/service-account.name: $argocd_sa
-type: kubernetes.io/service-account-token
-data:
-  ca.crt: $(yq e ".clusters.[0].cluster.certificate-authority-data" "$KUBECONFIG_KCP")
-  namespace: $(echo "$argocd_ns" | base64)
-  token: $(yq e ".users.[0].user.token" "$KUBECONFIG_KCP" | tr -d '\n' | base64)
----
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  clusterName: admin
-  name: $argocd_sa
-  namespace: $argocd_ns
-secrets:
-- name: $argocd_sa" >"$argocd_yaml"
-    KUBECONFIG="$KUBECONFIG_KCP" oc apply -f "$argocd_yaml" >/dev/null
-    rm "$argocd_yaml"
-    # </workaround>
-
-    sed -i'' -e 's:admin$:admin_kcp:g' "$KUBECONFIG_KCP"
-    KUBECONFIG="$KUBECONFIG_MERGED" argocd cluster add admin_kcp --name=kcp --yes >/dev/null 2>&1
+    KUBECONFIG="$KUBECONFIG_KCP" kubectl create ns kube-system  >/dev/null 2>&1
```

Comment:
I think that I have commented on that somewhere else. Why kube-system? I don't think we need to use anything else than the default namespace in a workspace but if we did "kube-system" would be the worst choice.

Reply:
When I execute command ` $ argocd cluster add "workspace.kcp.dev/current" --name=kcp --yes` to register the KCP cluster into ArgoCD, it reported error that there is no `kube-system` namespace. That's why I add this command to create the `kube-system` namespace

Reply:
you can use [--system-namespace](https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_cluster_add/) for that.

Reply:
Modified the command by using `--system-namespace` to specify `default` namespace

Reply:
Sorry, have you pushed the change? I still see kube-system

Reply:
@fgiloux sorry for late response. yes, I have pushed the change. and the script gitops.sh was change to openshift_dev_setup.sh. Maybe that's why you still see kube-system. If what I did is wrong against the PR change, please let me know. thanks